### PR TITLE
Fixing issues with dash seperators, sets, and duplicated exports

### PR DESCRIPTION
**Pull request by pashakondratyev**

### CODE REVIEW

File: src/codegen/ObjectCodeFile.ts [hunk 1]
```diff
@@ -6,7 +6,11 @@ import { getDocs } from "./getDocs.js";
 export const objectCodeGenerator = generatorFactory<IObjectDefinition>(
   async function() {
     const { typeName: { name }, docs } = this.def;
-    const fields = this.def.fields.map(f => `  ${f.fieldName}: ${this.getTypeForCode(f.type)};`)
+    const fields = this.def.fields.map(f =>
+      `  ${f.fieldName.includes("-") ? `"${f.fieldName}"` : f.fieldName}: ${
```
Comment (mfedderly) [inline src/codegen/ObjectCodeFile.ts:10]: is `-` the only character we have to worry about?

Comment (pashakondratyev) [reply src/codegen/ObjectCodeFile.ts]: Snake case is also valid, handling that

File: src/codegen/PackageCodeFile.ts [hunk 2]
```diff
@@ -28,10 +28,15 @@ export const packageIndexCodeGenerator = generatorFactory<
...
-    ).map(p => p.substring(this.def.packageName.length + 1).split(".")[0]);
+    const childPackagesSet = new Set();
+    this.codeGen.packages.forEach(p => {
+      if (p.startsWith(`${this.def.packageName}.`)) {
+        const childPackage = p.substring(this.def.packageName.length + 1).split(".")[0];
+        childPackagesSet.add(childPackage);
+      }
+    });
 
+    const childPackages = Array.from(childPackagesSet);
```
Comment (ericanderson) [inline src/codegen/PackageCodeFile.ts:39]: Am i missing something? I dont think this does anything

Comment (ericanderson) [reply src/codegen/PackageCodeFile.ts:39]: nm, im dumb

**Review by ericanderson — APPROVED** [review]

**Pull request merged**